app/main.py

```diff
@@ -1,96 +1,73 @@
-        self.surf = pygame.Surface((20, 40))
-        self.surf.fill((0, 255, 0))
-        self.rect = self.surf.get_rect()
-
-    def update(self):
-        pass
-
-
-class Wall(pygame.sprite.Sprite):
-    def __init__(self, center_pos: tuple):
-        super().__init__()
-        self.width = DEFAULT_OBJECT_SIZE
-        self.height = DEFAULT_OBJECT_SIZE
-        self.surf = pygame.Surface((self.width, self.height))
-        self.surf.fill((255, 255, 255))
-        self.rect = self.surf.get_rect(center=center_pos)
+if __name__ == "__main__":
+    # Setup for sounds
+    pygame.mixer.init()
 
-    @staticmethod
-    def create_centers_of_walls(field_size: tuple, wall_size: tuple):
-        center_width = wall_size[0] + wall_size[0] // 2
-        center_height = wall_size[1] + wall_size[1] // 2
-        centers = []
+    # Initialize program
+    pygame.init()
+    clock = pygame.time.Clock()
 
-        while center_height < field_size[1] - wall_size[1]:
-            while center_width < field_size[0] - wall_size[0]:
-                centers.append((center_width, center_height))
-                center_width += 2 * wall_size[0]
-            center_height += 2 * wall_size[1]
-            center_width = wall_size[0] + wall_size[0] // 2
+    # Background music
+    pygame.mixer.music.load(".\\music\\main_melody.mp3")
+    pygame.mixer.music.play(loops=-1)
 
-        return centers
+    # Set screen size
+    screen = pygame.display.set_mode([SCREEN_WIDTH, SCREEN_HEIGHT])
 
+    player = Player()
 
-player = Player()
+    walls = pygame.sprite.Group()
+    bombs = pygame.sprite.Group()
+    all_sprites = pygame.sprite.Group()
+    all_sprites.add(player)
 
-walls = pygame.sprite.Group()
-all_sprites = pygame.sprite.Group()
-all_sprites.add(player)
+    for wall_center in Wall.create_centers_of_walls(
+        (SCREEN_WIDTH, SCREEN_HEIGHT),
+        (DEFAULT_OBJECT_SIZE, DEFAULT_OBJECT_SIZE)
+    ):
+        wall = Wall(wall_center)
+        walls.add(wall)
+        all_sprites.add(wall)
 
-for wall_center in Wall.create_centers_of_walls(
-    (SCREEN_WIDTH, SCREEN_HEIGHT), (DEFAULT_OBJECT_SIZE, DEFAULT_OBJECT_SIZE)
-):
-    wall = Wall(wall_center)
-    walls.add(wall)
-    all_sprites.add(wall)
+    running = True
 
-running = True
+    while running:
+        for event in pygame.event.get():
+            if event.type == KEYDOWN:
+                if event.key == K_ESCAPE:
+                    running = False
 
-while running:
-    for event in pygame.event.get():
-        if event.type == KEYDOWN:
-            if event.key == K_ESCAPE:
+            elif event.type == QUIT:
                 running = False
 
-        elif event.type == QUIT:
-            running = False
-
-    player.update()
-
-    walls.update()
+        # Get all currently pressed
+        pressed_keys = pygame.key.get_pressed()
+        # Update the player sprite based on user key presses
+        player.update(pressed_keys, walls, bombs, all_sprites)
```

Comment:
It's better not to pass all these groups to the `update` method, but to store them in some `Engine` class. All sprites get reference on this class in constructor